awscli/testutils.py

```diff
@@ -1005,3 +1008,22 @@ def wait(self, check, *args, **kwargs):
     def _fail_message(self, attempts, successes):
         format_args = (attempts, successes)
         return 'Failed after %s attempts, only had %s successes' % format_args
+
+
+class YAMLStdoutDump(YAML):
+    """YAML class that can dump output as a string.
+
+    Taken from:
+
+    https://yaml.readthedocs.io/en/latest/example.html#output-of-dump-as-a-string
+
```

Comment:
nit; we probably don't need this newline or 1017. As a general rule (from PEP8) the closing `"""` should be on it's own line immediately following the last text.